tests/conftest.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+import sys
+import subprocess
+
+import pytest
+from six import StringIO
+
+from util import SphinxTestApp, path
+
+
+@pytest.fixture
+def app_params(request):
+    """
+    parameters that is specified by 'pytest.mark.sphinx' for
+    sphinx.application.Sphinx initialization
+    """
+    markers = request.node.get_marker("sphinx")
+    pargs = {}
+    kwargs = {}
+
+    if markers is not None:
+        # to avoid stacking positional args
+        for info in reversed(list(markers)):
+            for i, a in enumerate(info.args):
+                pargs[i] = a
+            kwargs.update(info.kwargs)
+
+    args = [pargs[i] for i in sorted(pargs.keys())]
+    return args, kwargs
+
+
+@pytest.fixture(scope='function')
+def app(app_params, make_app):
+    """
+    provides sphinx.application.Sphinx object
+    """
+    args, kwargs = app_params
+    app_ = make_app(*args, **kwargs)
+    yield app_
```

Comment:
There is no special meaning. yield and return is same meaning. (and next PR I will put teardown process here). Anyway, I will change this line into `return` for this PR.